src/debugpy/common/json.py

```diff
@@ -106,6 +108,15 @@ def of_type(*classinfo, **kwargs):
     def validate(value):
         if (optional and value == ()) or isinstance(value, classinfo):
             return value
+        elif (
+            isinstance(value, str)
+            and all(x in string.digits + "." for x in value)
+            and any(issubclass(x, numbers.Number) for x in classinfo)
+        ):
+            try:
+                return int(value)
+            except ValueError:
+                return float(value)
```

Comment:
One catch with this approach is that if the caller requests for `int`, they can now get a `float` instead, or vice versa. The former case is more dangerous, since the presence of the float will affect arithmetic down the line wherever that value flows. 

I would suggest that instead of doing a separate check and then a try/except series, to loop over the types in `classinfo`, and if they are a subtype of `Number`, then try to convert using that type object directly, and return if successful; if conversion fails, just move on to the next type. This will also take care of `bool` nicely.

Reply:
I have implemented the above logic